### PR TITLE
Handle date with iso format in input and output to simplify processing

### PR DESCRIPTION
**Pull request by alexandrebodin**
#### Description of what you did:
Handle date with iso format in input and output to simplify processing

#### My PR is a:

- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix
- [x] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:

- [ ] Admin
- [ ] Documentation
- [x] Framework
- [x] Plugin

#### Manual testing done on the following databases:

- [ ] Not applicable
- [ ] MongoDB
- [x] MySQL
- [ ] Postgres
- [x] SQLite

### CODE REVIEW

File: packages/strapi-hook-bookshelf/lib/buildDatabaseSchema.js [hunk 1]
```diff
@@ -383,6 +383,7 @@ const getType = ({ definition, attribute, name, tableExists = false }) => {
       return client === 'pg' ? 'double precision' : 'double';
     case 'decimal':
       return 'decimal(10,2)';
+    // TODO: split time types as they should be different
```
Comment (derrickmehaffy) [inline packages/strapi-hook-bookshelf/lib/buildDatabaseSchema.js]: Also timestamp vs datetime as timestamp does do some conversions hence why it is normally only used for fields like `created_at` and `updated_at`:

MySQL converts TIMESTAMP values from the current time zone to UTC for storage, and back from UTC to the current time zone for retrieval. (This does not occur for other types such as DATETIME.)

https://dev.mysql.com/doc/refman/8.0/en/datetime.html

Comment (alexandrebodin) [reply packages/strapi-hook-bookshelf/lib/buildDatabaseSchema.js]: Yes the next step will be to manage each of the types separatly. right now I'm aiming at being able to send and receive UTC iso dates so it is consistent ;)

Comment (derrickmehaffy) [reply packages/strapi-hook-bookshelf/lib/buildDatabaseSchema.js]: Perfect, that's what I thought I was going to look at splitting them also when develop get's merged. But I'll test this branch to make sure this patch is working

**Comment by derrickmehaffy** [comment]
@alexandrebodin @soupette 

Not sure if related (it was also doing this before but the Admin should never be sending the `created_at` and `updated_at` values. The database handles this, but in this case it breaks the adminUI for updates: 
image

I'm testing raw REST queries and GraphQL now

**Comment by derrickmehaffy** [comment]
REST Queries work even with an ISO formatted date headed to MySQL: 
image

**Comment by derrickmehaffy** [comment]
GraphQL also works fine: 
image

**Pull request merged**